spring-cloud-aws-autoconfigure/src/test/java/io/awspring/cloud/autoconfigure/cloudmap/integration/EcsCloudMapIntegrationTest.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2013-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.autoconfigure.cloudmap.integration;
+
+import java.time.Duration;
+import java.util.List;
+
+import io.awspring.cloud.autoconfigure.cloudmap.AwsCloudMapStoreClientCustomizer;
+import io.awspring.cloud.autoconfigure.cloudmap.CloudMapUtils;
+import io.awspring.cloud.autoconfigure.cloudmap.discovery.CloudMapDiscoveryClient;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+import software.amazon.awssdk.core.client.config.ClientOverrideConfiguration;
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.http.apache.ApacheHttpClient;
+import software.amazon.awssdk.services.servicediscovery.ServiceDiscoveryClient;
+
+import org.springframework.boot.BootstrapRegistry;
+import org.springframework.boot.BootstrapRegistryInitializer;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.WebApplicationType;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.test.system.OutputCaptureExtension;
+import org.springframework.context.ConfigurableApplicationContext;
+
+import static io.awspring.cloud.autoconfigure.cloudmap.CloudMapUtils.ECS;
+import static io.awspring.cloud.autoconfigure.cloudmap.CloudMapUtils.ECS_CONTAINER_METADATA_URI_V_4;
+
+/**
+ * Integration test for Ecs based Cloud Map registration.
+ *
+ * @author Hari Ohm Prasath
+ * @since 3.0
+ */
+@Testcontainers
+@ExtendWith(OutputCaptureExtension.class)
+public class EcsCloudMapIntegrationTest {
+
+	// Local stack container with Cloud Map and Ec2 services enabled
+	@Container
+	private static final LocalStackContainer localStackContainer = new LocalStackContainer(
+		DockerImageName.parse("localstack/localstack:1.1.0"))
```

Comment:
doesn't this need to be localstack-pro? I tried to use normal and it didn't work until I switched the image name. Maybe that changed since this PR was raised (current version is 3.2.0)